src/index.ts

```diff
@@ -61,8 +61,12 @@ const plugin: JupyterFrontEndPlugin<void> = {
         `jupyter-gallery API version out of sync, expected ${expectedVersion}, got ${data.apiVersion}`
       );
     }
-
     const title = data.title === 'Gallery' ? trans.__('Gallery') : data.title;
+
+    // hide the widget if no exhibits are configured
+    if (data.hideGalleryWithoutExhibits && !data.exhibitsConfigured) {
+      return;
+    }
     // add the widget to sidebar before waiting for server reply to reduce UI jitter
     if (launcher && isNewLauncher(launcher) && data.exhibitsConfigured) {
```

Comment:
I think we can then remove the previously added guard now:

```suggestion
    if (launcher && isNewLauncher(launcher)) {
```